src/plugins/simulator/visualizations/qt-opengl/qtopengl_camera.cpp

```diff
@@ -67,308 +64,213 @@ namespace argos {
-      /* Rotate around the local Y axis (Left)
-         The rotation matrix corresponding to this rotation is:
-         
-         | Fcos(a)-Usin(a)   L   Ucos(a)+Fsin(a) |
-         
-         where a is c_angle
-         L is the Left vector (local Y)
-         U is the Up vector (local Z)
-         F is the Forward vector (local X)
-      */
-      /* Calculate the new Up vector, given by:
-         Ucos(a)+Fsin(a)
-      */
-      /* Same, but faster than
-         cNewUp = Up * Cos(c_angle) + Forward * Sin(c_angle);
-      */
-      CVector3 cNewUp(Up);
-      cNewUp *= Cos(c_angle);
-      cNewUp += Forward * Sin(c_angle);
-      /* Check whether the rotation exceeds the limits */
-      if(cNewUp.GetAngleWith(CVector3::Z) > CRadians::PI_OVER_TWO) {
-         /*
-          * The rotation exceeds the limits
-          * The camera Up vector would form an angle bigger than 90 degrees with
-          * the global Z axis
-          */
-         /* We force the Up vector to lie on the XY plane */
-         Up.SetZ(0.0f);
-         Up.Normalize();
-         if(Forward.GetZ() < 0.0f) {
-            /* Forward was looking down, set it to -Z */
-            Forward = -CVector3::Z;
-         }
-         else {
-            /* Forward was looking up, set it to Z */
-            Forward = CVector3::Z;
-         }
-      }
-      else {
-         /* The rotation is OK */
-         Up = cNewUp;
-         Up.Normalize();
-         /* Now calculate the new Forward vector with a cross product
-            NOTE: the Left vector, being the rotation axis, remains
-            unchanged!
-         */
-         Forward = Left;
-         Forward.CrossProduct(Up).Normalize();
-      }
-   }
-
-   /****************************************/
-   /****************************************/
-
-   void CQTOpenGLCamera::SPlacement::RotateLeftRight(const CRadians& c_angle) {
-      /* Rotate around the local Z axis (Up)
-         The rotation matrix corresponding to this rotation is:
-         
-         | Fcos(a)+Lsin(a)   -Fsin(a)+Lcos(a)   U |
-         
-         where a is c_angle
-         L is the Left vector (local Y)
-         U is the Up vector (local Z)
-         F is the Forward vector (local X)
-      */
-
-      /* Calculate the new Forward vector, given by:
-         Fcos(a)+Lsin(a)
-      */
-      /* Same, but faster than
-         Forward = Forward * Cos(c_angle) + Left * Sin(c_angle);
-      */
-      Forward *= Cos(c_angle);
-      Forward += Left * Sin(c_angle);
-      Forward.Normalize();
-      /* Now calculate the new Left vector with a cross product
-         NOTE: the Up vector, being the rotation axis, remains
-         unchanged!
-      */
-      Left = Up;
-      Left.CrossProduct(Forward).Normalize();
-   }
-
-   /****************************************/
-   /****************************************/
-
-   void CQTOpenGLCamera::SPlacement::RotateLeftRight2(const CRadians& c_angle) {
```

Comment:
Why did you remove this code?

Reply:
It has been refactored into `CQTOpenGLCamera::Rotate`

Reply:
Why? At least, please keep the comments I wrote. It's not clear what the new `Rotate()` function does, partly because of the missing comments, partly because instead of accepting an angle as input, it accepts a `QPoint` (and then it immediately calculates an angle?).

Reply:
I have a note and a couple opinions here:
1. `CQTOpenGLCamera::Rotate` always took a `QPoint` as a single argument
2. All of the rotate function functions were only ever called from `CQTOpenGLCamera::Rotate` and only served the purpose of rotating the camera with respect to the user's mouse
3.  The comments in these functions were far too verbose and made the code hard to follow, perhaps such documentation should be on the website or a wiki?

Anyway, I already spent too much time working on this so if adding the comments back in and spiting the functions up is what it takes to get this merged, I will do it

Reply:
1. This is a convenience function for the Qt visualization.
2. Those functions make it possible to programmatically rotate the camera without reference to the mouse. I used them in the past to program panning in my videos. Plus, they're simple to read.
3. When I made that code, it took me 10 hours to get it right - it was quite a process for me. I don't think it's bad if the comments are verbose. I prefer to use a little bit of literate programming and not have to remember the equations, rather than having to guess them from the code.

I think it's not a problem if you stop working on this now. I'll have a thorough look at this code after August 11, and I'll probably changes things around anyway.

Reply:
Fixed in 9e922e763ec1d0a0ed639f0f5880a2a37d575623